docs/widgets/data_table.md

```diff
@@ -20,11 +24,143 @@ The example below populates a table with CSV data.
+
+Keys are important because cells in a data table can change location due factors like row deletion and sorting.
+Thus, using keys instead of coordinates allows us to refer to data without worrying about its current location in the table.
+
+Sometimes you wish to perform an operation on a cell/row/column based solely on coordinates, without regard for the data that is currently visible at those coordinates.
+In these cases you can make use of the [coordinate_to_cell_key][textual.widgets.DataTable.coordinate_to_cell_key] method to convert
+a coordinate to a _cell key_, which is a `(row_key, column_key)` pair.
+
+### Cursors
+
+You can retrieve the current coordinate of the cursor from the `cursor_coordinate` reactive attribute.
+Three types of cursor are supported: `cell`, `row`, and `column`.
+Change the cursor type by assigning to the `cursor_type` reactive attribute.
+
+=== "Column Cursor"
+
+    ```{.textual path="docs/examples/widgets/data_table_cursors.py"}
+    ```
+
+=== "Row Cursor"
+
+    ```{.textual path="docs/examples/widgets/data_table_cursors.py" press="c"}
+    ```
+
+=== "Cell Cursor"
+
+    ```{.textual path="docs/examples/widgets/data_table_cursors.py" press="c,c"}
+    ```
+
+=== "data_table_cursors.py"
+
+    ```python
+    --8<-- "docs/examples/widgets/data_table_cursors.py"
+    ```
+
+You can change the position of the cursor using the arrow keys, ++page-up++, ++page-down++, ++home++ and ++end++,
+or by assigning to the `cursor_coordinate` reactive attribute.
+
+### Updating data
+
+Cells can be updated in the `DataTable` by using the [update_cell][textual.widgets.DataTable.update_cell] and [update_cell_at][textual.widgets.DataTable.update_cell_at] methods.
+
+### Removing data
+
+To remove all data in the table, use the [clear][textual.widgets.DataTable.clear] method.
+To remove individual rows, use [remove_row][textual.widgets.DataTable.remove_row].
+The `remove_row` method accepts a `key` argument, which identifies the row to be removed.
+
+If you wish to remove the row below the cursor in the `DataTable`, use `coordinate_to_cell_key` to get the row key of
+the row under the current `cursor_coordinate`, then supply this key to `remove_row`:
+
+```python
+# Get the keys for the row and column under the cursor.
+row_key, _ = table.coordinate_to_cell_key(table.cursor_coordinate)
+# Supply the row key to `remove_row` to delete the row.
+table.remove_row(row_key)
+```
+
+### Fixed data
+
+You can fix a number of rows and columns in place, keeping them pinned to the top and left of the table respectively.
+To do this, assign an integer to the `fixed_rows` or `fixed_columns` reactive attributes of the `DataTable`.
+
+=== "Fixed data"
+
+    ```{.textual path="docs/examples/widgets/data_table_fixed.py"}
+    ```
+
+=== "data_table_fixed.py"
+
+    ```python
+    --8<-- "docs/examples/widgets/data_table_fixed.py"
+    ```
+
+In the example above, we set `fixed_rows` to `2`, and `fixed_columns` to `1`,
```

Comment:
Would be brilliant if the example also showed this. Maybe add 100 dummy rows to the table and then scroll to the bottom?

Reply:
Hmm, good idea! Will do that shortly.